lib/katello/tasks/clean_installed_packages.rake

```diff
@@ -32,30 +32,24 @@ namespace :katello do
 
     existing = Katello::HostInstalledPackage.select(:installed_package_id).uniq.pluck(:installed_package_id)
 
-    interrupted_run = false
     batch = 20_000
     until Katello::InstalledPackage.where('id not in (?)', existing).limit(1).count == 0
       begin
         deleted = Katello::InstalledPackage.where('id not in (?)', existing).where("id < #{batch}").delete_all
```

Comment:
would it make sense to also do .where("id > #{batch - 20_000}")  so it would actually keep going?

Reply:
I dunno, I was debating it a lot before choosing to do it this way. My main concern is that if the script is too clever, it will add bugs instead of subtracting. Right now the script is very robust against an infinite loop since it's monotonic, but if an insert "snuck in" underneath the batch window's floor, we would create an infinite loop. I don't think this is possible since IDs always increase, but I suppose you could craft a setup where this happens somehow near the end of the table.

Reply:
Also I was debating whether to print the batch size and say "re-run it but use this starting point instead", and let the script take the batch starting point as an argument. It can take some time for the script to run, but I think it's better to just re-run from the beginning unless the user has a slow machine.

Reply:
ah yeah that makes sense, ACK